master/reference/felix/configuration.md

```diff
@@ -117,40 +117,18 @@ The Kubernetes API datastore driver reads its configuration from Kubernetes-prov
-   with that `TyphaCN` value in their Common Name field.
-
-If both those parameters are set, a Typha certificate only has to match one of
-them; this is intended for when a deployment is migrating from using Common
-Name to using a URI SAN, to express identity.
-
 | Configuration parameter | Environment variable   | Description | Schema |
 | ----------------------- | ---------------------- | ----------- | ------ |
-| `TyphaCAFile`           | `FELIX_TYPHACAFILE`    | The full path to the certificate file for the Certificate Authorities that Felix trusts for Felix-Typha communications. | string |
-| `TyphaCertFile`         | `FELIX_TYPHACERTFILE`  | The full path to the certificate file for this Felix instance to use to connect to Typha. | string |
-| `TyphaCN`               | `FELIX_TYPHACN`        | If set, the Common Name that Typha's certificate must have. [Default: not set] | string |
-| `TyphaKeyFile`          | `FELIX_TYPHAKEYFILE`   | The full path to the private key file for this Felix instance to use to connect to Typha. | string |
-| `TyphaURISAN`           | `FELIX_TYPHAURISAN`    | If set, a URI SAN that Typha's certificate must have. [Default: not set] | string |
+| `TyphaCAFile`           | `FELIX_TYPHACAFILE`    | Use  this parameter to supply Felix with the path to the file containing the root certificate of the CA that issued the Typha server certificate. Configures Felix to trust the signature on the certificates provided by the Typha server. Example: `/etc/felix/ca.pem` | string |
+| `TyphaCertFile`         | `FELIX_TYPHACERTFILE`  | Path to the file containing the client certificate issued to Felix. Enables Felix to participate in mutual TLS authentication and identify itself to the Typha server. Example: `/etc/felix/cert.pem` | string |
+| `TyphaCN`               | `FELIX_TYPHACN`        | If set, the `Common Name` that Typha's certificate must have. If you have enabled TLS on the communications from Felix to Typha, you must set a value here or in `TyphaURISAN`. You can set values in both, as well, such as to facilitate a migration from using one to the other. It either matches, the communication succeeds. [Default: none] | string |
+| `TyphaKeyFile`          | `FELIX_TYPHAKEYFILE`   | Path to the file containing the private key of the Felix client certificate. Enables Felix to participate in mutual TLS authentication and identify itself to the Typha server. Example: `/etc/felix/key.pem` (optional) | string |
+| `TyphaURISAN`           | `FELIX_TYPHAURISAN`    | If set, a URI SAN that Typha's certificate must have. We recommend populating this with a unique [SPIFFE](https://github.com/spiffe/spiffe) ID. If you have enabled TLS on the communications from Felix to Typha, you must set a value here or in `TyphaCN`. You can set values in both, as well, such as to facilitate a migration from using one to the other. If either matches, the communication succeeds. [Default: none] | string |
```

Comment:
What do you mean by "unique" here?

Reply:
@neiljerram Perhaps you can tell me since you are closer to this functionality.

Reply:
I recommend deleting the "We recommend populating this with a unique [SPIFFE](https://github.com/spiffe/spiffe) ID." sentence.  Recommendations don't really belong here (reference), and the topic of SPIFFE is already very well covered in the new page that you've created.

That said, what I expect is:
- a SPIFFE ID that means "I am a Typha", and that is shared between multiple Typhas if there are multiple Typhas in the deployment; for example `spiffe://my-cluster.example.com/typha`
- a SPIFFE ID that means "I am a Felix", and that is shared between all the Felixes in the cluster; for example `spiffe://my-cluster.example.com/felix`.

From a technical perspective
- the Typha ID *needs* to be shared, because when a Felix connects to the Typha service, it doesn't know which Typha instance it will be connected to, yet whichever instance is selected needs to match that Felix's TyphaURISAN config
- similarly the Felix ID needs to be shared, because many Felixes will connect to a given Typha, and that Typha's ClientURISAN needs to match all the (valid) connecting Felixes.

But I think all of that fits better - if we think we need it at all - in the usage page.